src/Tasks/Microsoft.NET.Build.Tasks/CreateComHost.cs

```diff
@@ -20,19 +20,85 @@ public class CreateComHost : TaskWithAssemblyResolveHooks
         [Required]
         public string ClsidMapPath { get; set; }
 
+        public ITaskItem[] TypeLibraries { get; set; }
+
         protected override void ExecuteCore()
         {
             try
             {
+                if (!TryCreateTypeLibraryIdDictionary(TypeLibraries, out var typeLibIdMap))
+                {
+                    return;
+                }
                 ComHost.Create(
                     ComHostSourcePath,
                     ComHostDestinationPath,
-                    ClsidMapPath);
+                    ClsidMapPath,
+                    typeLibIdMap);
             }
             catch (ComHostCustomizationUnsupportedOSException)
             {
                 Log.LogError(Strings.CannotEmbedClsidMapIntoComhost);
             }
+            catch (TypeLibraryDoesNotExistException ex)
+            {
+                Log.LogError(Strings.TypeLibraryDoesNotExist, ex.Path);
+            }
+            catch (InvalidTypeLibraryIdException ex)
+            {
+                Log.LogError(Strings.InvalidTypeLibraryId, ex.Id.ToString(), ex.Path);
+            }
+        }
+
+        private bool TryCreateTypeLibraryIdDictionary(ITaskItem[] typeLibraries, out Dictionary<int, string> typeLibraryIdMap)
```

Comment:
Seems like we are passing in all the needed state, can we mark it `static`? If not, then what is the pattern for just grabbing properties off the instance instead of passing them in?

Reply:
The `Log` property is an instance property, so this can't be static.

Generally for MSBuild tasks, I like to pass the needed state in. Makes it easier for me to trace where things are used.